docs/docs/rest-api/public/api/v2/types/app.raml

```diff
@@ -0,0 +1,318 @@
+      like a custom Docker ENTRYPOINT. This args field may be used in place of
+      cmd even when using the default command executor.
+      This change mirrors API and semantics changes in the Mesos CommandInfo
+      protobuf message starting with version `0.20.0`.
+      Either `cmd` or `args` must be supplied.
+      It is invalid to supply both `cmd` and `args` in the same app.
+  backoffFactor?:
+    type: number
+    format: double
+    minimum: 1.0
+    description: |
+      Configures exponential backoff behavior when launching potentially sick
+      apps. This prevents sandboxes associated with consecutively failing tasks
+      from filling up the hard disk on Mesos slaves.
+      The backoff period is multiplied by the factor for each consecutive
+      failure until it reaches maxLaunchDelaySeconds.
+      This applies also to tasks that are killed due to failing too
+      many health checks.
+  backoffSeconds?:
+    type: integer
+    format: int32
+    minimum: 0
+    description: |
+      Configures exponential backoff behavior when launching potentially sick
+      apps. This prevents sandboxes associated with consecutively failing tasks
+      from filling up the hard disk on Mesos slaves.
+      The backoff period is multiplied by the factor for each consecutive
+      failure until it reaches maxLaunchDelaySeconds.
+      This applies also to tasks that are killed due to failing too
+      many health checks.
+  cmd?:
+    type: string
+    minLength: 1
+    description: |
+      The command that is executed.  This value is wrapped by Mesos via
+      `/bin/sh -c ${app.cmd}`.
+      Either `cmd` or `args` must be supplied.
+      It is invalid to supply both `cmd` and `args` in the same app.
+  constraints?:
+    type: array
+    # TODO: Unclear how to do an inline enum here.
+    items:
+      type: array
+      items: string
+  container?: !include appContainer.raml
+  cpus?:
+    type: number
+    format: double
+    description: |
+      The number of CPU shares this application needs per instance. This number does not have to be integer, but can be a fraction.",
+    minimum: 0.001
+  dependencies?:
+    type: array
+    items:
+      type: string
+      pattern: ^(\/?((\.{2})|([a-z0-9][a-z0-9\-.]*[a-z0-9]+)|([a-z0-9]*))($|\/))+$
+      description: |
+        A list of services upon which this application depends
+        An order is derived from the dependencies for performing start/stop and
+        upgrade of the application. For example, an application /a relies on the
+        services /b which itself relies on /c. To start all 3 applications, first
+        /c is started than /b than /a.
+  disk?:
+    type: number
+    description: |
+      How much disk space is needed for this application.
+      This number does not have to be an integer, but can be a fraction.
+    format: double
+  env?:
+    type: object
+    properties:
+      # TODO: How do we get secretRef.raml in here?
+      /.*/: string
```

Comment:
probably want the type of the properties to be `[ string | secretRef ]` or something like that